### PR TITLE
[Fix] Manage operational layers bugs

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements fixes for two bugs found with the `Manage operational layers` sample:
1. The Manage Layers settings pane showed as a blank screen when the it was accessed on Mac Catalyst. This was fixed by removing the `NaviagtionView` since it would cause a blank screen when used with the custom `sheet` implementation found in the Sample Viewer. This required adding UI components to mimic the original UI design. 
2. On iOS 15, the layer list rows couldn't be reordered by holding and dragging. This was fixed by replacing the "Done" button with an "Edit" button that will allow users to reorder the list items with the drag handles.
    - Note: The custom minus button implementation was kept because, without it, there is a bug where you are not able to remove list items that were removed and then re-added to the list. Also, the plus and minus buttons are different sizes on Mac Catalyst when the default implementation is used. Using the custom minus button does cause the row content on Mac Catalyst and iOS 15 to move right when the "Edit" button is pressed, but that cannot be helped. 

## Linked Issue(s)

- `swift/issues/4355`

## How To Test

- Tap the "Manage Layers" button.
- Tap "Edit" and drag the handles to reorder the operational layers.
- Tap "-" or swipe to remove an operational layer.
- Tap "+" to add a removed layer.
- Ensure it works on Mac Catalyst and iOS 15. 

## Screenshots

|Before|After|
|:-:|:-:|
|Screen Shot, Before|Screen Shot, After|

### CODE REVIEW

File: Shared/Samples/Manage operational layers/ManageOperationalLayersView.swift [hunk 1]
```diff
@@ -85,68 +81,74 @@ struct ManageLayersSheetView: View {
...
+                            Text(layer.name)
                         }
-                        Text(layer.name)
                     }
+                } header: {
+                    Text("Removed Layers")
                 }
             }
         }
+        .background(Color(.systemGray6))
```
Comment (yo1995) [inline Shared/Samples/Manage operational layers/ManageOperationalLayersView.swift:148]:
```suggestion
        .background(Color(.systemGroupedBackground))
```
I see what you are trying to do here. IMO the [semantic color](https://developer.apple.com/design/human-interface-guidelines/color#Platform-considerations) might be a better fit.

**Review by yo1995 — COMMENTED** [review]
This fix should work.

**Review by yo1995 — APPROVED** [review]

**Review by yo1995 — APPROVED** [review]

**Review by zkline101 — DISMISSED** [review]
Changes look good but just a few things that I noticed:
- Neither screenshot shows the "Edit" button 
- If you click the information button in this sample, it doesn't show the meta data. I am seeing this on this branch and v.next

**Comment by CalebRas** [comment]
Thanks Zac!
>  Neither screenshot shows the "Edit" button

The screenshots don't show the "Edit" button since it has already been pressed so that the drag handles are visible. Let me know if you want me to change that.

>  If you click the information button in this sample, it doesn't show the meta data. I am seeing this on this branch and v.next

This is related to another known bug, https://github.com/Esri/arcgis-maps-sdk-swift-samples/issues/80, which affects multiple samples.

**Comment by zkline101** [comment]
> The screenshots don't show the "Edit" button since it has already been pressed so that the drag handles are visible. Let me know if you want me to change that.

I think it is good to have one screenshot showing the drag handles and one where the edit button is visible. But not a huge deal I'll leave it up to @yo1995 .

**Comment by CalebRas** [comment]
> I think it is good to have one screenshot showing the drag handles and one where the edit button is visible. 

I updated one of them.

**Review by zkline101 — APPROVED** [review]

**Review by yo1995 — APPROVED** [review]

**Pull request merged**